### PR TITLE
Support class components in story snapshot tests

### LINKED ISSUE #240: x-dash test fails when component is a class
When we extend Preact's `Component` component class instead of using `x-interaction` for components that have state, and run `npm run jest`, we get this error:

```console
  ● @financial-times/x-topic-search › renders a default Topic Search Bar x-topic-search

    TypeError: Cannot call a class as a function
```

### PR DESCRIPTION
**Pull request by dan-searle**
Fixes #240

### CODE REVIEW

File: __tests__/snapshots.test.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
+/* eslint-disable no-unused-vars, react/react-in-jsx-scope */
```
Comment (dan-searle) [inline __tests__/snapshots.test.js]: Is there a better alternative to this?

File: __tests__/snapshots.test.js [hunk 2]
```diff
@@ -16,16 +18,16 @@ for(const pkg of packageDirs) {
...
 		const name = path.basename(pkg.name);
 
 		describe(pkg.name, () => {
 			for (const { title, data } of stories) {
-				for (const [ preset, options ] of Object.entries(presets)) {
+				for (const [preset, options] of Object.entries(presets)) {
 					it(`renders a ${preset} ${title} ${name}`, () => {
 						const props = { ...data, ...options };
-						const tree = renderer.create(component(props)).toJSON();
+						const tree = renderer.create(<Component {...props} />).toJSON();
```
Comment (i-like-robots) [inline __tests__/snapshots.test.js]: Could switch this to `h(Component, props)` to avoid the lint rule?

**Comment by dan-searle** [comment]
@i-like-robots thanks for the tip, I've made that change now.

File: __tests__/snapshots.test.js [hunk 3]
```diff
@@ -16,16 +17,16 @@ for(const pkg of packageDirs) {
 	const storiesDir = path.resolve(pkgDir, 'stories');
 
 	if(fs.existsSync(storiesDir)) {
-		const { package: pkg, stories, component } = require(storiesDir);
-		const { presets = {default: {}} } = require(pkgDir);
+		const { package: pkg, stories, component: Component } = require(storiesDir);
```
Comment (i-like-robots) [inline __tests__/snapshots.test.js]: I'm guessing the casing is no longer important without the JSX syntax?

Comment (dan-searle) [reply __tests__/snapshots.test.js]: Oh yes, of course.

**Pull request merged**